go/datas/remote_database_handlers.go

```diff
@@ -234,8 +234,38 @@ func handleRootPost(w http.ResponseWriter, req *http.Request, ps URLParams, rt c
 	d.PanicIfTrue(len(tokens) != 1, `Expected "current" query param value`)
 	current := hash.Parse(tokens[0])
 
-	if !rt.UpdateRoot(current, last) {
+	// Ensure that proposed new Root is present in cs
+	c := cs.Get(current)
+	d.PanicIfTrue(c.IsEmpty(), "Can't set Root to a non-present Chunk")
+
+	// Ensure that proposed new Root is a Map and, if it has anything in it, that it's <String, RefOfCommit>
+	v := types.DecodeValue(c, nil)
+	d.PanicIfTrue(v.Type().Kind() != types.MapKind, "Root of a Database must be a Map")
+	m := v.(types.Map)
+	if !m.Empty() && !isMapOfStringToRefOfCommit(m) {
+		panic(d.Wrap(fmt.Errorf("Root of a Database must be a Map<String, RefOfCommit>, not %s", m.Type().Describe())))
+	}
+
+	if !cs.UpdateRoot(current, last) {
 		w.WriteHeader(http.StatusConflict)
 		return
 	}
 }
+
+func isMapOfStringToRefOfCommit(m types.Map) bool {
+	mapTypes := m.Type().Desc.(types.CompoundDesc).ElemTypes
+	keyType, valType := mapTypes[0], mapTypes[1]
+	return keyType.Kind() == types.StringKind && (isRefOfCommitType(valType) || isUnionOfRefOfCommitType(valType))
```

Comment:
I tried that initially, and it does not handle this case. I think it's because this is actually a Union of Refs to something Commit-like, as opposed to a Ref to a Union of Commit-like things, and `isRefOfCommitType` handles only the latter.